packages/teams-js/src/internal/videoPerformanceStatistics.ts

```diff
@@ -0,0 +1,114 @@
+import { VideoFrameTick } from './videoFrameTick';
+
+export type VideoPerformanceStatisticsResult = {
+  effectId: string;
+  frameWidth: number;
+  frameHeight: number;
+  duration: number; // the duration in milliseconds that the data were collected
+  sampleCount: number; // the number of frames that were processed in the duration
+  distributionBins: Uint32Array; // an array that presents counts of frames that were finished in n milliseconds: distributionBins[frameProcessingDurationInMs]=frameCount. For example, distributionBins[10] = 5 means that 5 frames were processed in 10 milliseconds.
+};
+
+export class VideoPerformanceStatistics {
+  private static readonly initialSessionTimeoutInMs = 1000;
+  private static readonly maxSessionTimeoutInMs = 1000 * 30;
+
+  private currentSession: {
+    startedAtInMs: number;
+    timeoutInMs: number;
+    effectId: string;
+    frameWidth: number;
+    frameHeight: number;
+  };
+
+  private frameProcessingStartedAt: number;
+  private distributionBins: Uint32Array;
+  private sampleCount = 0;
+  private timeoutId: string;
+
+  public constructor(
+    distributionBinSize: number,
+    private report: (result: VideoPerformanceStatisticsResult) => void, // post event to the host
```

Comment:
I don't think I understand this comment. How does this `report` callback post the event to the host?

Reply:
Is it done using the app hosting repo internally? (Any internal code should not be discussed in this PR, we can discuss it async in chat)

Reply:
Ah I understand. I see now where we are instantiating the `VideoPerformanceMonitor` and passing `sendMessageToParent` to it, and then it's subsequently passed to this constructor.

Reply:
Yes, and I've updated the function name and the comments for better understanding